src/Home/SingleProduct.tsx

```diff
@@ -90,7 +78,9 @@ const ViewProduct = () => {
             </ul>
             <div className="flex space-x-4 mb-4">
               <button
-                onClick={addToCart}
+                onClick={() => {
+                  addToCart({ product, username, token, cartDispatch });
```

Comment:
### if token does not exists redirect to login the user add
>create function to redirect to login page if token does not exist when user try to add item to cart 

#4 #3  @20pa5a1210

Reply:
completed when user try to add to cart when not logged in, it shows popup message to login first